lineapy/api/api.py

```diff
@@ -211,16 +212,28 @@ def _try_write_to_db(value: object) -> Path:
                 k=7,
             )
         )
+        + ".pkl"
     )
+
+    artifact_storage_dir = UPath(options.safe_get("artifact_storage_dir"))
+    if not artifact_storage_dir.exists():
+        artifact_storage_dir.mkdir(parents=True)
+
+    filepath = artifact_storage_dir.joinpath(artifact_filename)
     try:
-        os.makedirs(filepath.parent, exist_ok=True)
-        with open(filepath, "wb") as f:
-            FilePickler.dump(value, f)
+        logger.debug(f"Saving file to {filepath.as_posix()} ")
+        with get_handle(
+            filepath,
+            mode="wb",
+            is_text=False,
+            storage_options=options.storage_options,
+        ) as handles:
+            FilePickler.dump(value, handles.handle)
```

Comment:
with remote handlers in place now, there is a possibility to more types of exception than just picklingError. Basic ones being network errors and auth/access errors (for buckets etc). All of these probably should be wrapped around ArtifactSaveException as well.

Reply:
We could wrap the whole thing in a general Exception block.